src/session/session_api.c

```diff
@@ -2124,6 +2125,18 @@ __wt_open_session(WT_CONNECTION_IMPL *conn, WT_EVENT_HANDLER *event_handler, con
         }
     }
 
+    /*
```

Comment:
The config parsing should go in session reconfigure. Reconfigure is a separate call and gets called from both SESSION.open and SESSION.reconfigure. Putting the config in reconfigure will allow you to open a session without debug, do some work, then enable-debug and evict content - while also allowing you to do so with open call as well.

Reply:
For reference look at reading config for ```ignore_cache_size``` and ```cache_cursors``` in ```__session_reconfigure()```